### PR TITLE
DOC: to_hdf() method docstring updated with dropna and index

### LINKED ISSUE #45030: DOC: index and dropna docs missing from to_hdf docstring
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `master` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/docs/dev/reference/api/pandas.DataFrame.to_hdf.html?highlight=to_hdf#pandas.DataFrame.to_hdf

### Documentation problem

- dropna
- index

are missing from the docstring

### Suggested fix for documentation

Add them

### PR DESCRIPTION
**Pull request by fanoway**
Resolves GitHub Issue #45030

- [x] closes #45030 
- [ ] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

**Comment by fanoway** [comment]
@github-actions pre-commit

**Comment by fanoway** [comment]
@github-actions pre-commit

**Comment by phofl** [comment]
Hey, thx for tackling this. I think there are a few places where this might be missing too. Should have mentioned in the issue

HDFStore.append and HDFStore.put are candidates.

**Comment by fanoway** [comment]
> Hey, thx for tackling this. I think there are a few places where this might be missing too. Should have mentioned in the issue
> 
> HDFStore.append and HDFStore.put are candidates.

Cheers, I'm getting a couple of failing checks in the github actions, not entirely sure hwat is going on as I only updated docstrings. Can you lend me some adivce cheers?

**Comment by phofl** [comment]
They are unrelated

**Comment by fanoway** [comment]
> 

Ok cheers. I fixed all the doc strings for the HDFStoe.put() and HDFStore.append() as they were missing quite a few parameters

**Comment by fanoway** [comment]
@github-actions pre-commit

File: pandas/io/pytables.py [hunk 1]
```diff
@@ -1228,8 +1246,27 @@ def append(
                 Table format. Write as a PyTables Table structure which may perform
                 worse but allow more flexible operations like searching / selecting
                 subsets of the data.
+        axes         : Optional, specify rows or columns
```
Comment (jreback) [inline pandas/io/pytables.py:1249]: don't add extra whitespace here (before the :)

**Review by jreback — CHANGES_REQUESTED** [review]
precommit is failing

**Comment by phofl** [comment]
@fanoway you can install pre-commit locally which will check a few styling things

File: pandas/io/pytables.py [hunk 2]
```diff
@@ -1103,16 +1103,34 @@ def put(
                 subsets of the data.
         append : bool, default False
             This will force Table format, append the input data to the existing.
+        complib: {‘zlib’, ‘lzo’, ‘bzip2’, ‘blosc’}, default ‘zlib’
+            pecifies the compression library to be used.
```
Comment (stdcout42) [inline pandas/io/pytables.py]: Typo 'specifies'

**Comment by phofl** [comment]
@fanoway Are you interested in continuing?

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. If interested in continuing please merge in main, address the comments and we can reopen

**Pull request closed without merging**